File: BackEnd/servicioAsignaturasCursos/src/main/java/co/unicauca/ra/model/Curso.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

package co.unicauca.ra.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

/**
*
 * @author ashle
 */
@Document(collection = "cursos")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Curso {
    @Id
    private String id; 
    private AsignaturaCurso asignatura; 
    private Docente docente;
    private Periodo periodo; 
}

File: FrontEnd/GestionRA-FrontEnd/src/app/Modelos/curso.ts
import { Docente } from "./docente";
import { Asignatura } from "./asignatura";
import { Periodo } from "./periodo";

export class Curso {

    id! : number;
    objAsignatura! : Asignatura;
    objDocente! : Docente;
    objPeriodo! : Periodo;

    constructor () {
        this.objAsignatura = new Asignatura();
        this.objDocente = new Docente();
        this.objPeriodo = new Periodo();
    }

}
